src/loaders/VolumeLoaderUtils.ts

```diff
@@ -76,3 +78,37 @@ export function estimateLevelForAtlas(spatialDimsZYX: number[][], maxAtlasEdge =
   }
   return levelToLoad;
 }
+
+function isEmpty(obj) {
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
I'm unfamiliar with this approach to testing for object emptiness. My instinct would be that `Object.keys(obj).length === 0` would do it, but it looks like this way might be more robust for objects with prototypes? Though `userData` is unlikely to have an inheritance chain.

Reply:
Yeah when I went to make userData optional, I just quickly googled it and found that `{}` is truthy, so ultimately I picked this one from lodash's implementation.

Reply:
JavaScript's strange approach to truth strikes again